feat: add service enrichment processor (#291)
kyma-project/opentelemetry-collector-components: processor/serviceenrichmentprocessor/README.md

# Service Enrichment Processor

The processor enriches the resource attribute `service_name` if it is not present. The processor follows a default priority
```yaml
    - "k8s.deployment.name",
	- "k8s.daemonset.name",
	- "k8s.statefulset.name",
	- "k8s.job.name",
	- "k8s.pod.name",
```

> Additionally one can define additional keys which would be prepended to the default priority. The keys added would follow the

Reviewer: ```suggestion
Additionally, you can define additional keys, which are prepended to the default priority. The added keys follow the priority in which they are defined.
```

> priority in which they are defined.

Reviewer: ```suggestion
```

# Service Enrichment Processor

| Status      |                             |
|-------------|-----------------------------|
| stability   | alpha: metrics,traces, logs |
| Code Owners | kyma-project/observability  |

The processor enriches the [service](https://opentelemetry.io/docs/specs/semconv/resource/#service) resource attribute if it is not present. The processor follows a default priority. Currently it

> has been implemented to enrich the service name based on the following keys:

Reviewer 2: Let's be more specific and write `resource attribute keys`?
